pkg/platform/runtime/setup/implementations/camel/runtime.go

```diff
@@ -18,12 +18,20 @@ func NewSetup(s *store.Store) *Setup {
 	return &Setup{s}
 }
 
-// ReusableArtifacts returns an empty, because camel installations cannot re-use and artifacts from previous installations
-func (s *Setup) ReusableArtifacts(_ artifact.ArtifactChangeset, _ store.StoredArtifactMap) store.StoredArtifactMap {
+// ReusableArtifacts returns an empty map, because camel installations cannot re-use and artifacts from previous installations
+// Only when the recipe is identical, it returns the stored artifact map.  This can happen when the executors director needs to be updated.
+func (s *Setup) ReusableArtifacts(changed artifact.ArtifactChangeset, stored store.StoredArtifactMap) store.StoredArtifactMap {
+	if len(changed.Added) == 0 && len(changed.Updated) == 0 && len(changed.Removed) == 0 {
+		return stored
+	}
 	return make(store.StoredArtifactMap)
 }
 
-func (s *Setup) DeleteOutdatedArtifacts(_ artifact.ArtifactChangeset, _, _ store.StoredArtifactMap) error {
+// DeleteOutdatedArtifacts deletes the entire installation directory, unless alreadyInstalled is not zero, which can happen when the executors directory needs to be re-generated.
+func (s *Setup) DeleteOutdatedArtifacts(_ artifact.ArtifactChangeset, _, alreadyInstalled store.StoredArtifactMap) error {
+	if len(alreadyInstalled) != 0 {
+		return nil
+	}
```

Comment:
This feels disjointed. Not a change for this PR, but it gives me a strong feeling that something is not modeled here. If my concern brings something to mind that isn't already in plans, please file a story.

Reply:
I agree: It is disjointed.  I think it is because this is (finally) designed for the "alt-builder"-use case.  In the "alt-case", the `alreadyInstalled` holds the *artifacts* that are already installed.  For the "camel"-use case, there are only two possibilities: 
1. None of the requested artifacts can be re-used
2. All of the requested artifacts can be re-used

I don't have an easy solution for it right now. Maybe we should file a ticket for it, discussing whether we have to clean up the "camel"-path?